torch/utils/hipify/hipify_python.py

```diff
@@ -36,6 +36,12 @@
 from .cuda_to_hip_mappings import CUDA_TO_HIP_MAPPINGS
 from .cuda_to_hip_mappings import MATH_TRANSPILATIONS
 
+# Allow hip files to exist in the same directory as cuda files.
+# This is useful when it is desired to avoid introducing a "hip" subdirectory
+# while there is no "cuda" subdirectory, which will affect the relationship
+# between device code and included header files.
+SUPPORT_SAME_DIR_HIPIFICATION = True
+
```

Comment:
I modified the call signature of the hipify function to be able to accept one more argument.  This global variable serves the sole purpose of maintaining backwards compatibility, mainly for detectron2, without relying on something like pytorch version  (so it can be tested prior to next pytorch release). If SUPPORT_SAME_DIR_HIPIFICATION exists, detectron2 will know that it can call hipify function setting the new allow_same_dir argument. If it doesn't, then the current approach with the ugly explicit copy of header files in the new hip subdirectories will be used instead.

Reply:
Detectron2 do check pytorch version to determine many features, that should be OK. I'm fine you testing the change locally but commit to detectron2 with something like `if torch_ver >= [1, 7]` which can't be true.

Reply:
Please remove this global var.  Then, in detectron2 code, you can use the inspect module to see if the allow_same_dir kwarg is present and react accordingly. 
```
>>> import inspect
>>> def func(a,b,c=42, *args, **kwargs): pass
>>> inspect.getargspec(func)
(['a', 'b', 'c'], 'args', 'kwargs', (42,))
```